testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jsonb/basic/JsonBindingDebugLoggingTest.java

```diff
@@ -0,0 +1,325 @@
+
+      LogCounter yassonExceptionLog = new LogCounter("Caused by: javax.json.bind.JsonbException", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter yassonStacktraceLog = new LogCounter("at org.eclipse.yasson", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter applicationExcpetionLog = new LogCounter(
+              "Caused by: java.lang.RuntimeException: "
+            + JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName(), true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      // perform request
+      WebTarget base = client.target(generateURL("/post"));
+      JsonBindingDebugLoggingItem wrongItem = new JsonBindingDebugLoggingItem();
+      wrongItem.setA(5);
+      Response response = base.request().post(Entity.entity(wrongItem,
+              MediaType.APPLICATION_JSON));
+
+      // check response
+      Assert.assertThat("Response message doesn't contains proper message",
+              response.readEntity(String.class), allOf(
+                      containsString("RESTEASY008200: JSON Binding deserialization error"),
+                      containsString(JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName()),
+                      containsString("javax.json.bind.JsonbException: ")
+              ));
+
+      // assert log messages after request in the server.log
+      Assert.assertThat("Application Exception should be logged",
+              applicationExcpetionLog.count(), is(1));
+      Assert.assertThat("RESTEasy exception should be logged",
+              resteasyExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception should be logged",
+              yassonExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception stacktrace should be logged",
+              yassonStacktraceLog.count(), greaterThan(0));
+
+      Assert.assertThat("There shouldn't be any error logs in server",
+              errorStringLog.count(), is(0));
+   }
+
+   /**
+    * @tpTestDetails Check exception during client receiving
+    * @tpSince RESTEasy 4.0.0.Beta7
+    *
+    * In this scenario the exception originates in JsonBindingProvider.readFrom.
+    * The exception is passed up the call stack to ClientResponse.readEntity which
+    * catches it as a RuntimeException.  The original exception is returned to the
+    * caller on the client side and this test logs the exception in the test's
+    * server.log for reference.
+    */
+   @Test
+   public void exceptionDuringClientReceive() throws Exception {
+      // perform request
+      WebTarget base = client.target(generateURL("/get/ok"));
+      Response response = base.request().get();
+
+      // count log messages before request
+      LogCounter errorStringLog = new LogCounter("ERROR",
+              true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter resteasyExceptionLog = new LogCounter(

```

Comment:
errorStringLog and resteasyExceptionLog are created, but not used

Reply:
This doesn't seem to be true any more.